hazelcast/src/main/java/com/hazelcast/instance/impl/ClusterTopologyState.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.instance.impl;
+
+import java.util.Objects;
+
+public class ClusterTopologyState {
```

Comment:
I would appreciate Javadoc here. Especially why we need `desiredNumberOfReplicas` etc.

Reply:
also please add state and intent difference in meaning in the javaDoc.

Reply:
class was removed in 96f2f61ac69d4ac11bcca2b032de9fdfb03269cf